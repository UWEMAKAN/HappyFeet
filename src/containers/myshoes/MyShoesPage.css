.MyShoesPage {
  display: flex;
  flex-direction: column;
  font-size: 1.6rem;
}

.MyShoesPage h2 {
  font-size: 3rem;
  font-weight: 600;
  color: #2c9cee;
  margin-bottom: 4rem;
}
