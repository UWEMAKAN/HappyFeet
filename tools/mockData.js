module.exports = {
  shoes: [
    {
      id: 1,
      title: 'Blue Snickers',
      brand: 'Addidas',
      image: './photos/addidas-blue-1-600x600.jpg',
      color: 'Blue',
      price: 325,
      discount: 20,
      rating: 4.4,
    },
    {
      id: 2,
      title: 'Black Snickers',
      brand: 'Others',
      image: './photos/others-black-1-600x600.jpg',
      color: 'Black',
      price: 295,
      discount: 25,
      rating: 4.3,
    },
    {
      id: 3,
      title: 'Black Snickers',
      brand: 'Others',
      image: './photos/others-black-2-600x600.jpg',
      color: 'Black',
      price: 325,
      discount: 25,
      rating: 4.3,
    },
    {
      id: 4,
      title: 'White Snickers',
      brand: 'Nike',
      image: './photos/nike-white-1-600x600.jpg',
      color: 'White',
      price: 450,
      discount: 20,
      rating: 4.6,
    },
    {
      id: 5,
      title: 'Snickers',
      brand: 'Addidas',
      image: './photos/addidas-green-1-600x600.jpg',
      color: 'Green',
      price: 400,
      discount: 25,
      rating: 4.5,
    },
    {
      id: 6,
      title: 'High Hills',
      brand: 'Others',
      image: './photos/others-black-hills-1-600x600.jpg',
      color: 'Black',
      price: 375,
      discount: 25,
      rating: 4.3,
    },
    {
      id: 7,
      title: 'White Snickers',
      brand: 'Nike',
      image: './photos/nike-white-2-600x600.jpg',
      color: 'White',
      price: 450,
      discount: 20,
      rating: 4.6,
    },
    {
      id: 8,
      title: 'Blue Snickers',
      brand: 'Addidas',
      image: './photos/addidas-blue-2-600x600.jpg',
      color: 'Blue',
      price: 400,
      discount: 25,
      rating: 4.5,
    },
    {
      id: 9,
      title: 'Black Leather Shoes',
      brand: 'Others',
      image: './photos/others-black-leather-1-600x600.jpg',
      color: 'Black',
      price: 425,
      discount: 25,
      rating: 4.3,
    },
    {
      id: 10,
      title: 'White Snickers',
      brand: 'Addidas',
      image: './photos/addidas-white-1-600x600.jpeg',
      color: 'White',
      price: 400,
      discount: 25,
      rating: 4.5,
    },
    {
      id: 11,
      title: 'Blue Snickers',
      brand: 'Others',
      image: './photos/others-blue-1-600x600.jpeg',
      color: 'Blue',
      price: 420,
      discount: 25,
      rating: 4.3,
    },
  ]
};
