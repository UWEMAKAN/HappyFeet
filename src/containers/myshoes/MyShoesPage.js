import React from 'react';
import propTypes from 'prop-types';
import MyShoeCard from '../../components/MyShoeCard/MyShoeCard';
import './MyShoesPage.css';
import NoResultsFound from '../../components/Common/NoResultsFound/NoResultsFound';

const MyShoesPage = ({ myShoes }) => {
  const shoes = myShoes.map((s) => (
    <MyShoeCard
      key={s.id}
      image={s.image}
      title={s.title}
      color={s.color}
      seller={s.seller}
      price={s.price}
    />
  ));
  return (
    <>
      {myShoes.length
        ? (
          <div className="MyShoesPage">
            <h2>My Shoes</h2>
            <div>
              {shoes}
            </div>
          </div>
        )
        : <NoResultsFound message="Nothing To Show" />}
    </>
  );
};

MyShoesPage.propTypes = {
  myShoes: propTypes.instanceOf(Array).isRequired
};

export default MyShoesPage;
