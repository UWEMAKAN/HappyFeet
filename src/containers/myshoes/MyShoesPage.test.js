import React from 'react';
import { shallow } from 'enzyme';
import MyShoesPage from './MyShoesPage';
import { shoes } from '../../../tools/mockData';

const renderMyShoesPage = (args) => {
  const defaultProps = {
    myShoes: []
  };
  const props = { ...defaultProps, ...args };
  return shallow(<MyShoesPage myShoes={props.myShoes} />);
};

describe('MyShoesPage Component', () => {
  test('should render MyShoesPage Component', () => {
    const myShoes = [shoes[0], shoes[1]];
    const wrapper = renderMyShoesPage({ myShoes });
    expect.assertions(5);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h2').text()).toEqual('My Shoes');
    expect(wrapper.getElements()[0].props.children.props.children.length).toBe(2);
    expect(wrapper.getElements()[0].props.children.props.children[0].type).toEqual('h2');
    expect(wrapper.getElements()[0].props.children.props.children[1].props.children.length).toBe(2);
  });
});
