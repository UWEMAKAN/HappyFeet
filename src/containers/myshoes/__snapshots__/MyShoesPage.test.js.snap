// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyShoesPage Component should render MyShoesPage Component 1`] = `ShallowWrapper {}`;
